src/org/gk/slicing/SlicingEngine.java

```diff
@@ -949,6 +1008,13 @@ private void extractReferencesToInstance(GKInstance instance) throws Exception {
                 pushToMap(tmp, sliceMap);
                 // Load all instances
 //                sourceDBA.loadInstanceAttributeValues(tmp);
+
+                // if the instance is an EntitySet, then populate the attributes passed to it.
+                if (tmp.getSchemClass().isa("EntitySet")) {
```

Comment:
It should not be checked and handled here. First it breaks the original logic flow and second the loading of attribute values will be extremely slow. The reason why there is sourceDBA.fastLoadInstanceAttributeValues(tmp) at next line.

Reply:
Ah — my mistake. I've removed this call in [9073362](https://github.com/reactome/CuratorTool/commit/90733623c17ba46684902632223c8e4a77bd361a) and will defer for now.